### DISCUSSION TITLE
FDR in proteomics & data filtering

### DISCUSSION THREAD

**Discussion by vdemichev** [vdemichev/DiaNN#1035] [DIA proteomics: in detail]
This is a note on the concept of false discovery rate (FDR) in proteomics, what do the q-values produced by the software mean, how to (and not to) estimate the 'real FDR' and how to compare different software tools.

### The concept of false discoveries in LC-MS proteomics and entrapment validation
First, I would like to discuss the definition of what it means that a particular precursor ion identification is false. One can consider three potential definitions:

* **F1**. Identification is false if none of the signals recorded by the mass spec in the matched spectrum originate from the same precursor ion species as reported by the software.

* **F2**. Identification is false if none of the signals recorded by the mass spec in the matched spectrum originate from the same or 'equivalent' precursor ion species as reported by the software. I will talk about what 'equivalent' means a bit later.

* **F3**. Identification is false if the sample does not contain the reported peptide or an 'equivalent' peptide.

F1 is the strictest definition, and FDR in the sense of F1, as will be clear from below, is almost never well controlled by the modern software. It might seem that F2 and F3 are quite similar, however the false discovery rates according to these definitions can in fact be dramatically different, depending on what the processing software is doing. The difference between F2 and F3 also has profound implications on the methods used to validate the FDR of the processing software.

Consider the following exaggerated example. The software searches a DIA experiment against a human database and detects 100k (i.e. hundred thousand) peptides as correctly matched to the spectra (according to F2) at 1% FDR, and 10k proteins matching these peptides, at 1% protein-level FDR. However, the total number of tryptic peptides corresponding to these proteins is ~750k, out of which ~700k were matched to something (true signal or pure noise) by the software. So what the software then does is report all those 700k peptides as identified. Question: is it OK for the software to do this, are those identifications useful?

Let's think what this all means. Let's assume the FDR estimates by the software are correct, i.e. out of those 10k proteins only about 1% - that is 100 - are false (according to F2). It's also a reasonable assumption that if a protein is detected, then all of its peptides are present in the sample at a non-zero copy number (might not be entirely true in reality, but let's assume for a second that it is). Then if the total number of tryptic peptides is comparable for true and false protein IDs, we would expect ~1% of the 700k peptides to be absent in the sample, i.e. the software does a fine job controlling FDR according to F3. However, the FDR according to F2 is not properly controlled at all: out of 700k peptides reported, maybe 110k-150k are matched to correct, albeit very noisy spectra, while >550k are likely matched to pure noise, that is, in this case, more than 70% of all peptide identifications are false according to F2. This is a problem because F3 FDR control on itself does not help to ensure reliability of quantities.

So is there any point in reporting those extra noisy peptide identifications? It does not really help with identifying proteins, it will not help with global FDR control. Such strategies make some practical sense only if used rather conservatively, ensuring that most of such identifications are quantitative. In DIA-NN, for example, specifically when analysing multiplexed data, the 'regular' precursor q-values have the meaning in the sense of F3, not F2. However, FDR in the sense of F2 is still controlled using dedicated 'channel' q-values, and quantitative accuracy is further controlled by the 'quantity quality' metric of QuantUMS.

The way the FDR rate of a software tool is often validated is via the so-called entrapment method, when the software is given also some extra peptides to search that are known to be absent in the sample. For example, the software can be asked to search a human cell line experiment against a database that includes both human and, say, _C. elegans_ protein sequences, or in silico generated shuffled sequences. Then the FDR according to F3 can be estimated given the proportion of 'decoy' (e.g. _C. elegans_ or shuffled) peptides or, respectively, proteins reported. Importantly, this experiment design tells nothing about F2 errors control, just F3.

So a software that includes protein information, in any way, in the process of decision-making on whether to report a particular peptide (in the sense of an amino acid (stripped) sequence) as identified, is, in a sense, cheating the 'entrapment' validation: peptide-level error rate estimates with the entrapment method are then no longer valid as controlling F2. Same in a situation when a decision on whether to report an individual precursor ion depends on the match confidence of other precursors sharing the same amino acid sequence. Here a software could do a very simple trick: calculate the q-values for stripped sequences and report all precursors (with all possible variable modifications and charge states) matched to noisy spectra as confident, if the respective stripped sequence is confident. And this will seemingly make the software produce very high precursor numbers at seemingly low entrapment FDR: but only controlling F3, not F2.

In addition, it is possible to guide run-specific decision-making using global confidence of identification. For example, if a precursor is identified globally with confidence, it can be 'safely' reported as identified in each run where it is matched to some noise - a regular entrapment validation will not detect this as a false ID, allowing the software to reach higher identification numbers and seemingly better completeness. Here there also can be a practical benefit in doing that, so long as this strategy is applied conservatively and the reported identifications are still largely quantitative: this is part of the idea of match-between-runs (MBR). But the entrapment method needs to be adapted to MBR, wherein some runs feature peptide digests from two species, whereas other runs feature peptide digests from a single species - and identifications of the other species' peptides or proteins in those are therefore known to be false.

**Summary**:

* The entrapment method can be 'cheated' in a number of ways. Precursor-level identification numbers at a given estimated entrapment FDR are only meaningful if the processing software applies no filters on peptide and protein identification confidence and does not create a strong dependence between reported identification confidence of different precursors in some other way.

* Protein-level entrapment FDR is more reliable, as long as the experiment involves a single acquisition or MBR is validated with an appropriate two-species experiment design. In the absence of such a design, only the global protein FDR can be validated.

**Most common mistakes** when validating FDR include leaving protein- or peptide-level FDR filters activated in the software configuration and then estimating precursor-level FDR, as well as looking at run-specific FDR oblivious of the fact that the analysis involved MBR (which might be called differently in different software). To highlight the magnitude of the effect of protein FDR filtering, it often reduces observed precursor-level entrapment FDR by 5- to 10-fold, with only a minor effect on the total numbers of reported precursors, therefore falsely making the impression that the software is 5x-10x more conservative in its FDR than it actually is. I note also that it could be that some peptide-level FDR filters are impossible to deactivate in some software, and they could even be completely invisible to the user (not possible to configure), and therefore comparisons of different black-box software tools using entrapment FDR at the precursor level are potentially unreliable. 

### Is the FDR reported by the software correct? Is the FDR estimated with an entrapment method correct?
This is a very interesting question, and it depends on the definition of 'equivalence' in F2. Suppose our sample is a peptide digest of highly purified bovine serum albumin (BSA), spiked in an _E. coli_ background. Suppose we do not care about _E. coli_ peptides and are searching only for BSA peptides. Suppose that BSA peptides are not similar to any _E. coli_ peptides in terms of sequence and hence the fragmentation spectrum. Suppose we know which peptides might bear PTMs and exclude them from search, as well as exclude any peptides with sites for modifications that often occur as technical artifacts, such as deamidation. Then whatever remaining BSA peptides are identified by the processing software, it is very likely that all these identifications are strictly correct, even according to F1, that is the respective spectra to which the peptides are matched indeed contain signals originating from the correct precursor ions.

Now, imagine a different situation. Same BSA, but now spiked into EV-enriched human plasma. The background is now very rich, with lots of peptides similar in sequence and fragmentation spectrum to the BSA peptides. However the software is still told to only look for BSA peptides, without excluding peptides that could bear modifications. The software will then likely use exactly the same approach for FDR estimation internally as in the first case, but will the reported identifications be strictly correct according to F1? Most likely not. It could be that the software reports, for some BSA peptide, a spectrum that actually originated from its modified (e.g. deamidated) form, or even a spectrum that originated from some human orthologue.

In the two cases above, it might be that the same BSA peptides will be reported by the software, at supposedly 1% FDR. But the FDR in the strict sense of F1, i.e. correctly matching to the spectrum that originates from the exact same peptidoform, could be very different in these scenarios. And it is challenging for the software to recognise the difference between these scenarios and adapt its FDR strategy (however DIA-NN does make some attempts to deal with this).

There are a variety of real world scenarios that lead to different effective FDR, in the sense of correct peptidoform matching, if filtering at the same 1% FDR as estimated by the software. For example, a tryptic digest of _E. coli_, a tryptic HeLa digest, a multi-enzyme HeLa digest with high degree of in-source fragmentation or a tryptic digest of a soil metaproteomics sample. So what FDR do the software tools report and is it 'correct'?

What the processing software does is employ a particular decoy model (e.g. mutated peptide sequences), score both targets and decoys and then choose a score threshold such that out of all peptides passing it, the proportion of decoys is less than a given margin, e.g. 1%. Does this reflect the actual FDR, in terms of correct identification of peptidoforms (F1)? As illustrated above, this will depend on the nature of the sample, but in most cases no, it does not.

Even for simple cell lysates, typical DIA software does not really control peptidoform FDR well. Rather it ensures that it can correctly distinguish peptides that are not sufficiently similar and can correctly distinguish proteins with low degree of sequence identity (i.e. not close paralogues or orthologues). The meaning of 'sufficiently' and 'low degree' here cannot be defined precisely, and likewise the 'equivalence' in the definition of F2 also cannot be defined precisely. The best we can do is assess the number of provably false discoveries using entrapment methods, e.g. using a human + _C. elegans_ database to analyse human samples, and then state that the FDR estimates reported by the software are good enough for distinguishing orthologues across the two species considered.

In DIA-NN 1.9, we take a step further and introduce a peptidoform scoring module that allows DIA-NN to report peptidoform q-values. Peptidoform scoring works similarly to regular FDR estimates, except a different decoy model is used and certain special characteristics of spectra are considered. It still does not control FDR in the strict sense of F1, e.g. it is not able to confidently localise phosphosites, but it can ensure confidence in the number of phosphates on the peptide.

When validating the FDR of the software, to compare different software tools, it makes sense to choose a reasonable entrapment decoy model. Below is a basic analysis of possible decoy models:

* Amino acids in the peptide are shuffled, leaving the C-terminal residue in place, in such a way that all masses of in silico predicted fragment ions become different. This is likely similar to what most peptide-centric DIA software tools do internally and is a sensible strategy.

* Amino acids in the peptide are shuffled, leaving the C-terminal residue in place, but without ensuring that the fragment masses are different. One can speculate that this can lead to sometimes overly conservative entrapment FDR estimates, making it look like DIA software is too optimistic in its FDR estimates. This strategy can in fact lead to some entrapment 'decoy' peptides having exactly the same predicted fragments as the corresponding target peptide, making it virtually impossible for DIA software to distinguish them, inflating the entrapment FDR estimate. I believe this is still an 'OK' strategy in practice though.

* Any of the above strategies, but with the number of decoys added to the database being fold-change more than there are targets. This is an unrealistic scenario (artificially inflated search space) that will fail to reflect the true software performance for two reasons. First, it will likely diminish the efficacy of the spectrum-centric module in DIA-NN and maybe other tools, reducing their advantage over tools that do not implement anything similar. Second, with a large number of decoys added, it becomes more likely that some of these decoys are very similar in sequence to true targets, and what is then being measured is effectively peptidoform-level FDR. Still, this is probably a suboptimal method for peptidoform FDR evaluation.

* Using an entrapment species, e.g. _C. elegans_ when analysing human samples. Whether the entrapment FDR estimates this strategy produces are less or more conservative than the internal decoy model used by the processing software depends on the degree of sequence identity between the target and entrapment proteomes.

* Two neighbouring amino acids are exchanged places. For example, P**EP**TIDE is in silico mutated to P**PE**TIDE. Then it is highly challenging for any software, DIA or DDA, to distinguish these without a special dedicated scoring module. Therefore, this decoy model will show very high apparent entrapment FDR for any modern proteomics software, probably exceeding, by far, the actual peptidoform-level FDR in its strictest sense (F1) in most experiments.

**Summary**. While one can define a 'true' false discovery rate as any kind of peptidoform misidentification, this is not what DIA (and DDA) software is really estimating. Rather that the software allows for certain tolerance with respect to sequence and fragmentation spectrum similarity between the reported precursor and the precursor from which the spectrum has actually originated. And this is perfectly fine for most 'regular proteomics' experiments. A peptidoform scoring module in DIA-NN 1.9 further allows also to control peptidoform identification confidence, with respect to mass shifts that are large enough, i.e. exceeding at least the span of the isotope envelope.

**Note**. DIA-NN does use several algorithms which treat its internal decoys somewhat differently from targets. In fact most DIA or DDA tools do, including training ML models (e.g. Percolator) to distinguish targets from decoys. The purpose of these algorithms is to achieve as high as possible true (entrapment-controlled) sensitivity. Some of these algorithms, however, make the internal FDR estimates of DIA-NN more optimistic, while some make them more conservative. Overall, the q-values eventually reported by DIA-NN usually correspond well to entrapment FDR using _C. elegans_ or _E. coli_ as an entrapment species, however the relationship between internal DIA-NN estimates and the entrapment FDR will inevitably vary between datasets. In almost all cases, the ratio of these will be in the 0.5 to 2.0 range (i.e. entrapment FDR will vary 0.5% to 2.0% at 1% reported FDR), which we believe is fine for practical purposes. While we can easily achieve a lot more precise match to entrapment FDR, at the cost of processing speed, we believe it unnecessary, given that the entrapment FDR itself is hardly an appropriate gold standard, given the considerations discussed above.

### Other ways to crudely validate identification correctness
It is sometimes possible to use methods other than entrapment to ensure that the software works 'well'.

* LFQbench-like experiments, wherein peptide digests from different species are mixed in known proportions, and the task of the workflow and the software is to recover those proportions as accurately as possible. The main purpose is to benchmark the quantification algorithms, but LFQbench also validates identification performance (F2), as misidentifications will typically result in grossly incorrect quantities.

* Comparing low-quality ultra-fast gradient/ultra-low load runs with high-quality runs (high load plus long gradient or gas-phase fractionation). The similarity between such peptide characteristics as measured signal intensity or aligned retention time, between low- and high-quality runs, indirectly validates the identifications (F2) in the low-quality runs.

* Spike-in mixtures of known peptides, e.g. to validate phosphosite localisation algorithms.

### Q-values and posterior error probabilities (PEPs)
The q-values in DIA-NN mean that if you filter the output at q-value < _x_, then based on the internal decoy model that DIA-NN uses the proportion of false discoveries (in the sense of F2) in the filtered output is not expected to exceed _x_. This however does not directly reflect the confidence in individual identifications. For this, DIA-NN calculates PEP, which is always more conservative than the q-value. To increase the chance that identifications in question are actually quantitative, it does make sense to also filter the DIA-NN output using PEP and PG.PEP columns in the .parquet main report.

### Filtering the DIA-NN main report (.parquet)
Below is a summary of best practices for filtering the DIA-NN main report. The filtering is done trivially in R or Python, in just several lines of code. That being said, for many experiments it is OK to just use the pg_matrix reported by DIA-NN, without a need to work in R or Python. The threshold ranges listed below offer quite a lot of flexibility. Usually, a good idea is to apply a filter that has only a minor effect on the total number of identifications, and the exact threshold which achieves that will depend on the dataset.

DIA-NN can analyse experiments with MBR (recommended, unless you use a project-specific DIA-based spectral library) and without MBR. MBR can also be 'imitated' manually, by first creating a spectral library from DIA data, potentially from just a subset of runs, and then using this library to analyse the whole experiment. In the latter case, the filtering applied should be the same as if MBR were used. When creating DIA-based spectral libraries with DIA-NN, precursor FDR can be set to any value between 1% and 5%, although 1% is currently the recommended default setting.

* It always makes sense to filter the data using global precursor (Global.Q.Value) and protein group (Global.PG.Q.Value) q-values. If using MBR, use the 'Lib.' q-values instead of 'Global.' ones. If using a DIA-based library that might contain a significant number of proteins not detectable in your experiment (e.g. the library is based on the bulk sample, while you are analysing single cells), makes sense to filter using both 'Lib.' and 'Global.' q-values. Never apply a filter to 'Global.' q-values without applying at least as strict filter to the respective 'Lib.' q-values. Reasonable filtering thresholds for most applications are 0.01 - 0.05 (i.e. 1% to 5%). Lib.Q.Value filter is by default applied at 1% when doing MBR. Lib.PG.Q.Value filter is not applied by default to accommodate workflows that either do not feature complete protein annotation or focus on peptides rather than proteins (e.g. when working with PTMs).

* It always makes sense to filter also using run-specific q-values. On the precursor level (Q.Value), 0.01 is the reasonable threshold (and is used by default). On the protein group level (PG.Q.Value), thresholds in the range 0.01 - 0.05 are reasonable. Never apply a filter to run-specific q-values that is more strict than the filter applied to 'Lib.' q-values, except for one special case: if you would like to evaluate the proteomic depth achieved in specific runs, it is OK to analyse with MBR, with precursor FDR set to 1% in the GUI, and then count precursor and protein identification at 1% run-specific q-values, without applying a global protein q-value filter.

* It often does make sense to apply PEP and PG.PEP filters. Reasonable thresholds are 0.01 to 0.5 (not a typo).

* When using QuantUMS (which is always recommended), makes sense to filter using Quantity.Quality, PG.MaxLFQ.Quality and sometimes the average values of those across the experiment. Reasonable thresholds are 0.01 to 0.8. Sometimes, a filter based on Empirical.Quality (reasonable thresholds 0.01 to 0.7) can also be helpful.

* When analysing plexDIA (multiplexed) samples, it is essential to apply the Channel.Q.Value filter, with reasonable thresholds 0.01 to 0.5 (not a typo). PG.Q.Value is also channel-specific and reasonable thresholds are in the range 0.01 to 0.05, similarly to the label-free scenario. It is also particularly important to apply some of the QuantUMS filters listed above.

* When analysing PTMs or aiming to ensure peptidoform confidence, always apply the Global.Peptidoform.Q.Value or (with MBR) Lib.Peptidoform.Q.Value filter instead. Reasonable thresholds are 0.01 to 0.05. Optionally, apply run-specific Peptidoform.Q.Value filtering, although often this is not really necessary. Reasonable thresholds are 0.01 to 0.5. When PTM localisation is important, for PTMs declared as variable modifications, apply PTM.Site.Confidence and/or (with MBR) Lib.PTM.Site.Confidence (thresholds 0.5 to 0.99). Alternatively, can also work with confidence levels for individual sites, which are also reported, or just use the phosphosites_90 and phosphosites_99 matrices.

**Comment by percolator** [comment]
Hi Vadim,

Thank you for supporting the DIA-NN software and sharing your thoughts on reporting false discovery rates (FDR) in proteomics with DIA-NN. I have a few points to consider:

* Avoid PSM-Level FDR Reporting: Reporting FDR at the peptide level is more meaningful in proteomics than at the PSM level. For more on this, refer to [this paper](https://pubs.acs.org/doi/10.1021/acs.jproteome.2c00282).

* Error Rate Example Clarification: Your initial "exaggerated example" suggests using the FDR of one type of list (e.g., proteins) to report the error rates of another list (e.g., PSMs). This is not advisable and can lead to misinterpretation of the data.

* At some points in the text there is the suggestion that challenging identifications should be discounted for by considering "equivalent peptides" as they are difficult for the software to interpret. That will lead to problems downstream.

* Drawbacks of DIA-NN Reporting:
   - Decoy Hits: DIA-NN does not output decoy hits, making it difficult for others to validate your assumptions.    
   - Combining Identifications: It is problematic to merge identifications from different runs of DIA-NN together. Transferring q-values between different lists of identifications is, again, not something you could do as it leads to incorrect FDR estimations. If you reported decoys the FDR calculation could be redone after list merger, however, there it currently could not be done without some intricate manipulations of your reported list of results (as we made with Yasset's paper (can't share this to the public yet)).

I look forward to further discussing these points and improving our methods for accurate FDR reporting.

**Reply by vdemichev** [reply]
Hi Lukas, many thanks for sharing your thoughts! Fully agree with all points. This is also why I decided to put this opinion write-up on github, as it allows anyone to comment & the whole thing becomes more informative.

> Avoid PSM-Level FDR Reporting: Reporting FDR at the peptide level is more meaningful in proteomics than at the PSM level. For more on this, refer to [this paper](https://pubs.acs.org/doi/10.1021/acs.jproteome.2c00282).

* Yes, my point too, also that PSM-level FDR is challenging to validate if the software scores PSMs originating from the same sequence 'jointly' and not in an independent fashion.

> Error Rate Example Clarification: Your initial "exaggerated example" suggests using the FDR of one type of list (e.g., proteins) to report the error rates of another list (e.g., PSMs). This is not advisable and can lead to misinterpretation of the data.

* I wanted to actually argue against that. What some software tools do is automatically filter the output using protein FDR. Nothing wrong with it per se, but it invalidates any precursor/peptide-level entrapment validation.

> At some points in the text there is the suggestion that challenging identifications should be discounted for by considering "equivalent peptides" as they are difficult for the software to interpret. That will lead to problems downstream.

Apologies, I guess I did not phrase things the way I intended. My point is that our q-values / PEPs currently don't control F1 error rate but only control F2 for some definition of 'equivalence', but we don't know this definition. For example, if, hypothetically, we had some dataset with known ground truth, we could calculate the proportion of F1 errors and we could also calculate the proportion of F2 errors for a given (non-transitive) equivalence definition. Say, we can define "PEPTIDE 1 is equivalent to PEPTIDE 2 if their masses differ by less than 5 Da and at least 80% of theoretical fragments of PEPTIDE 1 are also theoretical fragments of PEPTIDE 2. Then our F2 error rate will be the proportion of PSMs such that the precursor in question is matched to a spectrum from an equivalent peptide. That was the idea. 

> Decoy Hits: DIA-NN does not output decoy hits, making it difficult for others to validate your assumptions

We now actually have everything implemented which allows to output the full list of decoys. So this functionality will be added soon.

**Reply by heejongkim** [reply]
Hi Vadim,

Does 1.9.2 have an option to output the full list of decoy and everything or any upcoming version may have the functionality? 

Thanks.

**Comment by GeorgWa** [comment]
Hi Vadim,

Very interesting read and a good summary. There are some things I still want to implement myself like peptide level FDR. I also agree that it should be possible to get outputs only with precursor FDR and peptide FDR etc. applied for entrapment analysis.

Another important point is that false discovery rates have the temptation to be interpreted in an absolute fashion (F1), disregarding the context of the experiment. This means that you can have highly confident identifications based on limited signal (only MS1, few fragments) if the context of the library allows this (multiplexing or extremly good RT calibration). I think this can even be entrapment controlled (alltough good entrapments without leakage are challenging in these cases). This does not mean that peptides where identfied based on some absolute standard (like requiring a full b- and y-series for de-novo) but it was only possible in the context of the experiment and library. Therefore, peptide-centric and spectrum-centric search engines will show very different behavior in such cases.

I find the idea of peptidoform scoring quite intriguing. Can you share how this works in DIA-NN in terms of the features and decoy model?

I'm looking forward to further discussions!

**Reply by vdemichev** [reply]
Hi Georg, great to hear from you! Btw, congrats on the very impressive work with AlphaDIA! 
Peptidoform scoring in DIA-NN is indeed based on a separate decoy model, we will describe this in a publication, and while we are very happy with how it performs at the moment, we might still adjust the way it works by the time.

**Comment by oliverschilling** [comment]
Dear Vadim,

thanks for providing DIANN to the community. We frequently perform test searches using „added shuffled sequences“ (same number of entries for real and shuffled entries) as a rough and quick estimate of search engine performance.  We have always been very pleased with the performance of DIANN, always seeing much less than 1 % shuffled entries (at 1 % FDR) at precursor/peptide and protein level. This has been consistently the case for different types of samples, mass specs (and users...).

Reporting of decoys will of course be highly appreciated. 

I wanted to take this opportunity and ask a question about FDR determination at peptide vs protein level: Peptides from "real" proteins should aggregate on the proteins from which they have generated; e.g. 50 peptides from a specific collagen. On the other hand, I would assume that "decoy peptides" are more randomly distributed across decoy proteins. Is it possible that this imbalance has an impact on FDR determination at protein level? For example: 990 "real" peptide IDs + 10 "decoy" peptide IDs might represent (theoretically, if the above assumption is correct and the "real" peptides aggregate on fewer proteins than the decoy ones) 300 "real" protein IDs + 9 "decoy" peptide IDs. Would this consideration impact the protein-level FDR or am I overlooking something?

Thanks,

Oliver

**Reply by vdemichev** [reply]
Hi Oliver,

Many thank for the feedback!

Indeed, this peptides per protein distribution seems like a major factor.

> 990 "real" peptide IDs + 10 "decoy" peptide IDs might represent (theoretically, if the above assumption is correct and the "real" peptides aggregate on fewer proteins than the decoy ones) 300 "real" protein IDs + 9 "decoy" peptide IDs. 

Exactly what we observe!

> Would this consideration impact the protein-level FDR or am I overlooking something?

The way we do it in DIA-NN is via target protein vs decoy protein competition. That is, rather than trying to infer protein FDR from just target peptide IDs we also take decoy peptide IDs in the account, which is enabled by having a decoy peptide query for each target peptide query. So DIA-NN just scores all target & decoy proteins in a particular way, treating them completely equally, and then the estimated FDR is the proportion of decoy proteins at a given score threshold.

Best,
Vadim

**Comment by Harry-young-0** [comment]
Really nice description. 

Following on from BSPR last week, I wondered if you caught Andy Jones's talk on FDR in phosphopeptides? Are you planning on adding an Alanine-Phospho decoy for FDR as a feature or did you have any thoughts on how this might fit into a DIANN phosphoproteomics workflow?

Cheers,

Harry

**Comment by vdemichev** [comment]
Hi Harry,

> Andy Jones's talk on FDR in phosphopeptides

Yep, his and colleagues work is the basis how we validate it :) 

> Are you planning on adding an Alanine-Phospho decoy for FDR

We may. For now, that's how we test it internally. There'are also different ways you can define a q-value (real vs estimated) based on localisation probabilities. We adopted a very strict definition when developing 2.2 and made sure it's output is in line with decoy AA validation (we use different ones) on a number of diverse datasets. 

That said, we decided not to include such automatic validation into 2.2 itself, reason being that it may be quite confusing to the user and also add unnecessary complication (since we pass validation on all data we examined, no reason to do this per se). However, the user can always run dia-nn with an empirical lib while declaring phospho as variable on any AAs of your choise, e.g. on STYA. The site_report then allows to compare observed vs expected false localisation rates, validating DIA-NN's performance on specific data. You may want to also try with --loc-no-h3po4 in this case. There's also an excellent review on this: https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/10.1002/pmic.202300088, also on considerations when using losses. 

Best,
Vadim

**Reply by Harry-young-0** [reply]
Fab, Thanks Vadim, really cool stuff, I'll try adding STYA to that variable mod and see what it looks like.

This review looks like exactly what I need to read haha!

Thanks again,

Harry
